diff --git a/node_modules/await-sync/mod.js b/node_modules/await-sync/mod.js
index 867ee90..0a31830 100644
--- a/node_modules/await-sync/mod.js
+++ b/node_modules/await-sync/mod.js
@@ -24,6 +24,7 @@ function createWorker (signal) {
     worker.postMessage({ port: remotePort, code: source, ab }, [remotePort])
 
     return function runSync (...args) {
+      Atomics.store(int32, 0, 0)
       // Send the arguments to the worker thread
       localPort.postMessage(args)
       // Wait for the worker thread to send the result back
diff --git a/node_modules/await-sync/worker.js b/node_modules/await-sync/worker.js
index fdb4f16..a507084 100644
--- a/node_modules/await-sync/worker.js
+++ b/node_modules/await-sync/worker.js
@@ -24,8 +24,7 @@ addEventListener('message', async evt => {
     }
   }
 
-  const blob = new Blob([code], { type: 'text/javascript' })
-  const url = URL.createObjectURL(blob)
+  const url = "data:text/javascript," + encodeURIComponent(code)
   const { default: fn } = await import(url)
 
   port.onmessage = async function onmessage (evt) {
@@ -50,3 +49,16 @@ addEventListener('message', async evt => {
     write2(u8)
   }
 })
+
+export {};
+if (typeof process !== "undefined") {
+  // You can't console.log() anything since all I/O goes THROUGH THE MAIN THREAD
+  // which is blocked by Atomics.wait(). Thus, we need to write somewhere else.
+  const { appendFileSync } = await import("node:fs");
+  process.on("uncaughtExceptionMonitor", e => {
+    appendFileSync("error.log", e.stack + "\n");
+  });
+  process.on("unhandledRejection", e => {
+    appendFileSync("error.log", e.stack + "\n");
+  });
+}
